extensions/mailer/runtime/src/main/java/io/quarkus/mailer/Mail.java

```diff
@@ -163,20 +163,46 @@ public Mail setFrom(String from) {
     }
 
     /**
-     * @return the reply-to address.
+     * @return the reply-to address. In the case of multiple addresses, the comma-separated list is returned, following
+     *         the https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.2 recommendation. If no reply-to address has been
+     *         set, it returns {@code null}.
      */
     public String getReplyTo() {
-        return replyTo;
+        if (replyTo == null || replyTo.isEmpty()) {
+            return null;
+        }
+        return String.join(",", replyTo);
     }
 
     /**
      * Sets the reply-to address.
+     * <p>
+     * If you add multiple reply-to addresses, it adds {@code replyTo} to the list.
      *
      * @param replyTo the address to use as reply-to. Must be a valid email address.
      * @return the current {@link Mail}
+     * @see #setReplyTo(String[])
      */
     public Mail setReplyTo(String replyTo) {
-        this.replyTo = replyTo;
+        if (this.replyTo == null) {
+            this.replyTo = new ArrayList<>();
+        }
+        this.replyTo.add(replyTo);
+        return this;
+    }
+
+    /**
+     * Sets the reply-to addresses.
+     *
+     * @param replyTo the addresses to use as reply-to. Must contain valid email addresses, must contains at least
+     *        one address.
+     * @return the current {@link Mail}
+     */
+    public Mail setReplyTo(String... replyTo) {
```

Comment:
I don't really like the fact that the method is a `set` but in the end you add the addresses.

I think I would rather have a real `set` for now and see if people wants an `add` version later.

Reply:
I'd rather have only `get` methods in `Mail` and have a Builder with the set/add methods.

Reply:
Well, this is orthogonal to what I'm saying. I just want a method called `set` to really do a set.

Reply:
That was to avoid introducing a breaking change. Yes, the method should be add. 

So we can have both methods and have a deprecation cycle, or accept the breaking change.